tests/test_utils/test_progressbar_rich.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import time
+
+import mmengine
+
+
+class TestRichProgressBar:
+
+    def test_start(self):
+        # single task
+        prog_bar = mmengine.RichProgressBar()
+        prog_bar.add_task(100)
+        assert len(prog_bar.bar.tasks) == 1
+        assert prog_bar.bar.tasks[0].total == 100
+        assert prog_bar.infinite is False
+        del prog_bar
+        # multi tasks
+        prog_bar = mmengine.RichProgressBar()
+        for i in range(5):
+            prog_bar.add_task(10)
+        assert len(prog_bar.bar.tasks) == 5
+        del prog_bar
+        # without total task num
+        prog_bar = mmengine.RichProgressBar()
+        prog_bar.add_task(None)
+        assert prog_bar.infinite is True
+        del prog_bar
```

Comment:
There are three more cases that should be tested:
1. If an infinite task has been added, no more finite tasks could be added
2. If some finite tasks has been added, infinite task should not be added